connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/TestPlugins.java

```diff
@@ -143,16 +176,35 @@ public static void assertAvailable() throws AssertionError {
     public static List<String> pluginPath() {
```

Comment:
The existing structure of this class bakes in the assumption that each plugin only appears once on the plugin path, and that the common use-case plugin path (returned by this method) is then valid. This change would make this method return an invalid plugin path, with flakey behavior when loading the duplicated plugin (which plugin gets loaded would be determined by iteration order over the PLUGIN_JARS hashmap).

There are a couple of ways out:
1. avoid tackling this now, and separate the two plugins with different class names
2. have this method pick one of the duplicates to drop deterministically, so that the Plugins class doesn't have undefined loading behavior.
3. allow/deny some of these plugins from being included in this default plugin path, and keep some plugins back for the more specific tests
4. remove this method, and have PluginsTest explicitly include the needed plugins in each test, and/or a default list to include if none are specifically requested.

Reply:
Yes, but plugins are now no longer unique by name. The `test.plugins.ReadVersionFromResource` plugin class is now provided by two different plugin jars, and when the DelegatingClassLoader attempts to load it, it is arbitrary which one is chosen.

For example, if you set up Plugins with this path, and load the `test.plugins.ReadVersionFromResource`, will you get v1 or v2? I believe it depends on iteration order in the hashmap, and so happens to be deterministic here but not in general.

Currently, Connect does not have first class support for plugin paths with duplicate plugins due to this ambiguity, so I don't think this method should return a plugin path which is unsupported.